subnet_sharing.tf

```
@@ -1,9 +1,28 @@
-module "subnet_sharing" {
-  source                = "github.com/schubergphilis/terraform-aws-mcaf-subnet-sharing?ref=v0.3.0"
-  count                 = var.share_public_subnets || var.share_private_subnets ? 1 : 0
-  name                  = var.name
-  prepend_resource_type = var.prepend_resource_type
-  private_subnet_arns   = var.share_private_subnets ? toset(aws_subnet.private[*].arn) : []
-  public_subnet_arns    = var.share_public_subnets ? toset(aws_subnet.public[*].arn) : []
-  tags                  = merge(var.tags, var.subnet_sharing_custom_tags)
+locals {
+  sharing_private_subnet_arns = var.share_private_subnets ? aws_subnet.private[*].arn : []
+  sharing_public_subnet_arns  = var.share_public_subnets ? aws_subnet.public[*].arn : []
+
+  sharing_tags = merge(var.tags, var.subnet_sharing_custom_tags)
+}
+
+resource "aws_ram_resource_share" "subnet_sharing" {
+  count = var.share_public_subnets || var.share_private_subnets ? 1 : 0
+
+  name                      = "${var.prepend_resource_type ? "resource-share-" : ""}subnets-${var.name}"
```

```suggestion
  name                      = "subnets-${var.name}"
```

Shall we make this a bit cleaner and remove `var.prepend_resource_type`?

---

This PR is just to include the module code, so no changes for now for discussion.